api/pkg/apis/v1alpha1/providers/stage/materialize/materialize.go

```diff
@@ -559,3 +566,13 @@ func updateObjectMeta(objectMeta model.ObjectMeta, inputs map[string]interface{}
 	}
 	return objectMeta
 }
+
+func checkCatalog(catalog *model.CatalogState) bool {
+	if catalog.Spec == nil {
+		return false
+	}
+	if catalog.Spec.CatalogType == "instance" || catalog.Spec.CatalogType == "solution" || catalog.Spec.CatalogType == "target" || catalog.Spec.CatalogType == "catalog" {
```

Comment:
Consider case ignore?

Reply:
We still use case sensitive check in the later switch case. Do you want me change both to case sensitive?

Reply:
@LeiQL , let's create a bug. We need to iterate all codes and change them to case insensitive compare.